src/rules/curlyRule.ts

```diff
@@ -54,76 +95,150 @@ export class Rule extends Lint.Rules.AbstractRule {
-                && !isStatementBraced(node.elseStatement)) {
+    public visitDoStatement(node: ts.DoStatement) {
+        this.validateIterationStatement(node, ts.SyntaxKind.DoKeyword);
+        super.visitDoStatement(node);
+    }
 
-            // find the else keyword to place the error appropriately
-            const elseKeywordNode = node.getChildren().filter((child) => child.kind === ts.SyntaxKind.ElseKeyword)[0];
+    public visitWhileStatement(node: ts.WhileStatement) {
+        this.validateIterationStatement(node, ts.SyntaxKind.WhileKeyword);
+        super.visitWhileStatement(node);
+    }
 
-            this.addFailure(this.createFailure(
-                elseKeywordNode.getStart(),
-                node.elseStatement.getEnd() - elseKeywordNode.getStart(),
-                Rule.ELSE_FAILURE_STRING
-            ));
+    private validateIfStatement(node: ts.IfStatement) {
+        const isIfSameLine = areOnSameLine(node, node.thenStatement);
+        const isIfSingleLine = isStatementSingleLine(node.thenStatement);
+        const isIfBraced = isStatementBraced(node.thenStatement);
+
+        this.validateStatementPositioning(node, ts.SyntaxKind.IfKeyword, isIfSameLine, isIfSingleLine, isIfBraced);
+
+        if (node.elseStatement != null &&
+            node.elseStatement.kind !== ts.SyntaxKind.IfStatement) {
+            const elseKeywordNode = getElseKeywordNode(node);
+            const isElseSameLine = areOnSameLine(elseKeywordNode, node.elseStatement);
+            const isElseSingleLine = isStatementSingleLine(node.elseStatement);
+            const isElseBraced = isStatementBraced(node.elseStatement);
+
+            this.validateStatementPositioning(node,
+                                              ts.SyntaxKind.ElseKeyword,
+                                              isElseSameLine,
+                                              isElseSingleLine,
+                                              isElseBraced);
         }
+    }
 
-        super.visitIfStatement(node);
+    private validateIterationStatement(node: ts.IterationStatement, kind: ts.SyntaxKind) {
+        const isSameLine = areOnSameLine(node, node.statement);
+        const isSingleLine = isStatementSingleLine(node.statement);
+        const isBraced = isStatementBraced(node.statement);
+
+        this.validateStatementPositioning(node, kind, isSameLine, isSingleLine, isBraced);
     }
 
-    public visitDoStatement(node: ts.DoStatement) {
-        if (!isStatementBraced(node.statement)) {
-            this.addFailureForNode(node, Rule.DO_FAILURE_STRING);
+    private validateStatementPositioning(node: ts.Node,
+                                         kind: ts.SyntaxKind,
+                                         isSameLine: boolean,
+                                         isSingleLine: boolean,
+                                         isBraced: boolean) {
+
+        if (((this.getOptions().length === 0) || this.hasOption(OPTION_ALL)) && !isBraced) {
```

Comment:
parse options one time when rule is loaded and store flags on the class.  Should also treat [true] as [true, "all"]